packages/ui-keyring/src/Keyring.ts

```diff
@@ -82,6 +82,18 @@ export class Keyring extends Base implements KeyringStruct {
     return pair.toJson(password);
   }
 
+  public async backupAccounts (pairs: KeyringPair[]): Promise<KeyringPair$Json[]> {
+    const accountPromises = pairs.map((pair) => {
+      return new Promise((resolve) => {
+        this._store.get(accountKey(pair.address), resolve);
+      });
+    });
+
+    const accounts = await Promise.all(accountPromises);
+
+    return accounts as KeyringPair$Json[];
```

Comment:
It would be good to actually return the full JSON as a string here, not the pairs. This means when encrypting/decrypting, it all happens here.

Reply:
I've thought it through. Returning stringified JSON does make sense, but it would lose information of the return type which makes maintenance tough.
Do you insist on this?

Reply:
It is easier actually, that way the extension doesn't need to worry about the specific format at all. So from here the keyring could already return a full-encrypted version.

Reply:
I get your point there.

Reply:
Quite happy if we encrypt and don't stringify and pass this type in/out -

https://github.com/polkadot-js/common/blob/master/packages/util-crypto/src/json/types.ts#L14

It is fine if the extension calls stringify. (And probably aligns closer to what we have with account export as well)